fixed float issue

joerivanarkel opened the pull request.

teusvanarkel commented on Interpreter/Values/Float.cs:
```diff
@@ -16,12 +16,12 @@ public Float(object value)
     public override BaseValue OperatedBy(Token _operator, BaseValue other)
     {
-        if (Value is not double && other.Value is not double)
+        if (Value is not float && other.Value is not float)
```
override Value is maybe not necessary, try to use the base Value

teusvanarkel commented on Interpreter/Values/Variable.cs:
```diff
@@ -4,40 +4,48 @@
     public ILogger Logger { get; set; }
 
-    public Variable(string value, ILogger logger)
+    public Variable(string value, ILogger logger, InterpreterFactory interpreterFactory)
```
Maybe move the logic to a method

teusvanarkel commented on Interpreter/Values/Variable.cs:
```diff
@@ -4,40 +4,48 @@
+           Value = result.Value;
+        }
+        else
+        {
```
Remove duplicate call to the symbol table

teusvanarkel commented on Interpreter/Values/Variable.cs:
```diff
@@ -4,40 +4,48 @@
     {
-        ValueNode = (IValueNode) SymbolTable.Instance(logger).Get(value);
-        if (ValueNode.Value.Value is null)
+        var ResultNode = SymbolTable.Instance(logger).Get(value);
```
Lower case

teusvanarkel commented on Interpreter/Values/Variable.cs:
```diff
@@ -4,40 +4,48 @@
 public class Variable : BaseValue, IValue
 {
-    public IValueNode ValueNode { get; set; }
+    public IValueNode? ValueNode { get; set; }
```
Remove...not used

teusvanarkel commented on Interpreter/Values/Variable.cs:
```diff
@@ -4,40 +4,48 @@
+            case Type charType when Value.GetType() == typeof(char):
                 return new Char(Value, Logger).OperatedBy(_operator, other);
         }
         Logger.Log("No TypeCode found", this.GetType().Name, Common.Enum.LogType.ERROR);
```
Removed logging just before error....log the error!

joerivanarkel commented:
**Summary**:
- Moved Logger and Factory to BaseInterpeter
- Moved Logger and Value to BaseVaule
- Added more possible nodes to variables
- Started on Validation methods per entity. #60 
- Fixed Float and Character lexing errors
